BankAPILibrary/GiniBankAPILibrary/Tests/GiniBankAPILibraryTests/DocumentServiceTests.swift

```
@@ -201,5 +174,63 @@ final class DocumentServicesTests: XCTestCase {
+              let lineItems = compoundExtractions.lineItems,
+              let firstLineItem = lineItems.first else {
+            XCTFail("Missing compound extractions or line items")
+            return
+        }
 
+        let filteredCompoundExtractions = ["lineItems": [firstLineItem]]
+        let pay4Keys = ["amountToPay", "iban", "reference", "paymentRecipient"]
+        let filteredExtractions = extractionResult.extractions.filter { extraction in
+            pay4Keys.contains(extraction.name ?? "")
+        }
+
+        defaultDocumentService.submitFeedback(for: document,
+                                              with: filteredExtractions,
+                                              and: filteredCompoundExtractions) { result in
+            switch result {
+                case .success:
+                    self.verifySuccessfulSubmission(amountToPay: amountToPay,
+                                                    amountToPayFromLoadedFeedback: amountToPayFromLoadedFeedback,
+                                                    expectation: expectation)
+                case .failure:
+                    break
+            }
+        }
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func verifySuccessfulSubmission(amountToPay: String,
+                                    amountToPayFromLoadedFeedback: String,
+                                    expectation: XCTestExpectation) {
+        DispatchQueue.main.async {
+            guard let httpBody = self.sessionManagerMock.extractionFeedbackBody,
+                  let json = try? JSONSerialization.jsonObject(with: httpBody, options: .mutableContainers) as? [String: [Extraction]],
+                  let extractions = json["extractions"],
+                  let amountToPayFromHttpBody = extractions.first(where: { $0.name == "amountToPay" })?.value else {
+                print("json data malformed or missing amount")
```

[Clarification]: Shall we fail test here instead of using print?

---

Yes we can. I will do it. Thank you!